barman/recovery_executor.py

```diff
@@ -1807,13 +1810,327 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+        Move a single directory or file to its destination directory
+
+        :param barman.infofile.SyntheticBackupInfo backup_info: the synthetic
+            backup info file
+        :param barman.infofile.Tablespace: the tablespace to be moved
+        :param dict[str,str]|None tablespaces_mapping: a tablespace
+            name -> location map (for relocation)
+        """
+        for file_or_dir in os.listdir(source):
+            if file_or_dir not in exclude_dirs:
+                file_or_dir_path = os.path.join(source, file_or_dir)
+                try:
+                    shutil.move(file_or_dir_path, destination)
+                except shutil.Error:
+                    output.error(
+                        "Destination directory '%s' must be empty." % destination
+                    )
+                    output.close_and_exit()
+
+    def _get_backup_chain_path(self, backup_info):
+        """
+        Get the path of the backup chain, from the full backup to the specified
+        incremental backup.
+
+        :param barman.infofile.LocalBackupInfo backup_info: The incremental backup
+        """
+        return reversed(
+            [backup.get_data_directory() for backup in backup_info.walk_to_root()]
+        )
+
+    def _prepare_destination(self, dest_dir):
+        """
+        Prepare the destination of a backup data/tablespace.
+
+        This method is responsible for removing a directory if it already
+        exists and for ensuring the correct permissions on the directory
+
+        :param str dest_dir: destination directory
+        """
+        # Remove a dir if exists. Ignore eventual errors
+        shutil.rmtree(dest_dir, ignore_errors=True)
+        # create the dir
+        mkpath(dest_dir)
+        # Ensure the right permissions to the destination directory
+        # chmod 0700 octal
+        os.chmod(dest_dir, 448)
+
+    def _retry_handler(self, dest_dirs, attempt):
+        """
+        Handler invoked during a combine backup in case of retry.
+
+        The method simply warn the user of the failure and
+        remove the already existing directories of the backup.
+
+        :param list[str] dest_dirs: destination directories
+        :param int attempt: attempt number (starting from 0)
+        """
+        output.warning(
+            "Failure combining backups using pg_combinebackup (attempt %s)", attempt
+        )
+        output.warning(
+            "The files created so far will be removed and "
+            "the combine process will restart in %s seconds",
+            "30",
+        )
+        # Remove all the destination directories and reinit the backup
+        for d in dest_dirs:
+            self._prepare_destination(d)
+
+    def _fetch_remote_status(self):
+        """
+        Gather info from the remote server.
+
+        This method does not raise any exception in case of errors,
+        but set the missing values to None in the resulting dictionary.
+        """
```

Comment:
Missing `:return:`.

Reply:
dict[str, str|bool] is this ok? Usually when i used docstrings and annotations we use that typing module with List[str, Union[str,bool]].